crates/extensions/c8y_mapper_ext/src/operations/firmware_update.rs

```diff
@@ -215,13 +216,28 @@ mod tests {
         // Simulate firmware_update cmd metadata message
         mqtt.send(MqttMessage::new(
             &Topic::new_unchecked("te/device/main///cmd/firmware_update"),
+            "{}",
+        ))
+        .await
+        .expect("Send failed");
+
+        // Simulate firmware_update twin firmware message
+        mqtt.send(MqttMessage::new(
+            &Topic::new_unchecked("te/device/main///twin/firmware"),
             r#"{"name":"firmware", "version":"1.0"}"#,
         ))
         .await
         .expect("Send failed");
 
         // Validate if current installed firmware message is sent
-        assert_received_contains_str(&mut mqtt, [("c8y/s/us", "115,firmware,1.0,")]).await;
+        assert_received_contains_str(
+            &mut mqtt,
+            [(
+                "c8y/inventory/managedObjects/update/test-device",
+                r#"{"c8y_Firmware":{"name":"firmware","version":"1.0"}}"#,
+            )],
+        )
+        .await;
```

Comment:
This part should move to `inventory.rs` as it tests actually our `twin` feature.